feature_engine/creation/cyclical.py

```diff
@@ -0,0 +1,140 @@
+    variables.
+
+    There are some feature that are cyclic by nature. One example of this are
+    the hours of a day or the months of a year. In both cases the higher values of
+    a set of data are closer to the lower values of that set.
+
+    For example the month October (10) is closer to month January (1) than
+    to month February (2).
+
+    To check an explanation about this:
+    http://blog.davidkaleko.com/feature-engineering-cyclical-features.html
+
+    The CyclicalTransformer() works only with numerical variables. But does not
+    allow null values on any row of original column
+
+    A list of variables can be passed as an argument. Alternatively, the
+    transformer will automatically select and transform all numerical
+    variables.
+
+    Parameters
+    ----------
+    variables : list, default=None
+        The list of numerical variables that will be transformed. If None, the
+        transformer will automatically find and select all numerical variables.
+    max_values: dict(str, Union)
+        A dictionary that maps the natural maximum or a variable. Useful when
+        the maximum value is not present in the dataset.
+    drop_original: bool, default=False
+        Use this if you want to drop the original columns from the output.
+
+
+    Methods
+    -------
+    fit:
+        This transformer does not learn parameters.
+    transform:
+        Apply the CyclicalTransformer transformation.
+    fit_transform:
+        Fit to data, then transform it.
+
+    References
+    ----------
+    http://blog.davidkaleko.com/feature-engineering-cyclical-features.html
+    """
+
+    def __init__(
+            self, variables: Union[None, int, str, List[Union[str, int]]] = None,
+            max_values: Optional[Dict[str, Union[int, float]]] = None,
+            drop_original: Optional[bool] = False
+    ) -> None:
+        self.variables = _check_input_parameter_variables(variables)
+        self.max_values = max_values
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Learns the max_value of each of the numerical variables.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y : pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame.
+        ValueError:
+            - If some of the columns contains NaNs.
+            - If some of the mapping keys are not present in variables.
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+        _check_contains_na(X, self.variables)
```

Comment:
BaseNumericalTransformer already checks for na, so this line is duplicated / not needed:
https://github.com/solegalli/feature_engine/blob/master/feature_engine/base_transformers.py#L62